diff --git a/gladius/tests/unittests/ApplicationMCPAdapter_Rollback_tests.cpp b/gladius/tests/apitests/ApplicationMCPAdapter_Rollback_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/ApplicationMCPAdapter_Rollback_tests.cpp
rename to gladius/tests/apitests/ApplicationMCPAdapter_Rollback_tests.cpp
diff --git a/gladius/tests/unittests/ApplicationMCPAdapter_tests.cpp b/gladius/tests/apitests/ApplicationMCPAdapter_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/ApplicationMCPAdapter_tests.cpp
rename to gladius/tests/apitests/ApplicationMCPAdapter_tests.cpp
diff --git a/gladius/tests/integrationtests/GladiusLib_tests.cpp b/gladius/tests/apitests/GladiusLib_tests.cpp
similarity index 100%
rename from gladius/tests/integrationtests/GladiusLib_tests.cpp
rename to gladius/tests/apitests/GladiusLib_tests.cpp
diff --git a/gladius/tests/integrationtests/MCP_tests.cpp b/gladius/tests/apitests/MCP_Integration_tests.cpp
similarity index 100%
rename from gladius/tests/integrationtests/MCP_tests.cpp
rename to gladius/tests/apitests/MCP_Integration_tests.cpp
diff --git a/gladius/tests/unittests/MCP_tests.cpp b/gladius/tests/apitests/MCP_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/MCP_tests.cpp
rename to gladius/tests/apitests/MCP_tests.cpp
diff --git a/gladius/tests/apitests/main.cpp b/gladius/tests/apitests/main.cpp
new file mode 100644
index 0000000..7af96e6
--- /dev/null
+++ b/gladius/tests/apitests/main.cpp
@@ -0,0 +1,7 @@
+#include <gtest/gtest.h>
+
+auto main(int argc, char ** argv) -> int
+{
+    ::testing::InitGoogleTest(&argc, argv);
+    return RUN_ALL_TESTS();
+}
diff --git a/gladius/tests/apitests/testdata.h b/gladius/tests/apitests/testdata.h
new file mode 100644
index 0000000..a8ae05d
--- /dev/null
+++ b/gladius/tests/apitests/testdata.h
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <string>
+namespace gladius_integration_tests
+{
+    struct FileNames
+    {
+        static auto constexpr SimpleGyroid = "testdata/SimpleGyroid.3mf";
+        static auto constexpr F360 = "testdata/F360.3mf";
+        static auto constexpr VariableVoronoi = "testdata/variable_voronoi.3mf";
+        static auto constexpr Implicit3mf = "testdata/RadialRadiator.3mf";
+    };
+}
\ No newline at end of file
diff --git a/gladius/tests/apitests/testhelper.cpp b/gladius/tests/apitests/testhelper.cpp
new file mode 100644
index 0000000..f43bce6
--- /dev/null
+++ b/gladius/tests/apitests/testhelper.cpp
@@ -0,0 +1,70 @@
+#include "testhelper.h"
+
+#include <array>
+#include <iostream>
+
+#ifdef WIN32
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif
+#endif
+#include <lodepng.h>
+#include <fmt/format.h>
+
+#include "BitmapLayer.h"
+
+namespace gladius_integration_tests
+{
+    std::optional<std::filesystem::path> findGladiusSharedLib()
+    {
+        std::array<std::string_view, 2> const pathsToTest{
+          "./", "../../src/api/GladiusLib_component/Implementations/Cpp"};
+        std::array<std::string_view, 2> const binaryNames{"gladiuslib.dll", "gladiuslib.so"};
+
+        for (auto const & path : pathsToTest)
+        {
+            try
+            {
+                auto const absolutePath = std::filesystem::canonical(path);
+                for (auto const & binaryName : binaryNames)
+                {
+                    if (std::filesystem::exists(absolutePath / binaryName))
+                    {
+                        return absolutePath / binaryName;
+                    }
+                }
+            }
+            catch (std::exception const & e)
+            {
+                std::cout << "Exception thrown during search of shared library: " << e.what()
+                          << "\n";
+            }
+        }
+
+        return {};
+    }
+
+    void saveBitmapLayer(std::filesystem::path const & filename,
+                         std::vector<float> & data,
+                         unsigned int width_px,
+                         unsigned int height_px)
+    {
+        std::vector<unsigned char> image;
+        image.resize(data.size());
+
+         std::cout << "exporting bitmap to " << filename << "\n";
+        std::transform(data.begin(),
+                       data.end(),
+                       image.begin(),
+                       [](auto value) { return static_cast<unsigned char>(value * 1000.f); });
+
+        auto const error = lodepng::encode(filename.string(), image, width_px, height_px, LodePNGColorType::LCT_GREY);
+
+        if (error)
+        {
+            throw std::runtime_error(
+               fmt::format("encoder error {} : {}", error , lodepng_error_text(error)));
+        }
+    }
+
+}
diff --git a/gladius/tests/apitests/testhelper.h b/gladius/tests/apitests/testhelper.h
new file mode 100644
index 0000000..a2b5d16
--- /dev/null
+++ b/gladius/tests/apitests/testhelper.h
@@ -0,0 +1,15 @@
+#pragma once
+#include <filesystem>
+#include <optional>
+#include <vector>
+
+
+namespace gladius_integration_tests
+{
+    [[nodiscard]] std::optional<std::filesystem::path> findGladiusSharedLib();
+    
+     void saveBitmapLayer(std::filesystem::path const & filename,
+                         std::vector<float> & data,
+                         unsigned int width_px,
+                         unsigned int height_px);
+}
diff --git a/gladius/tests/unittests/ColorExport_Integration_tests.cpp b/gladius/tests/integrationtests/ColorExport_Integration_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/ColorExport_Integration_tests.cpp
rename to gladius/tests/integrationtests/ColorExport_Integration_tests.cpp
diff --git a/gladius/tests/unittests/DualContouringOctree_tests.cpp b/gladius/tests/integrationtests/DualContouringOctree_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/DualContouringOctree_tests.cpp
rename to gladius/tests/integrationtests/DualContouringOctree_tests.cpp
diff --git a/gladius/tests/unittests/DualContouringStlExporter_tests.cpp b/gladius/tests/integrationtests/DualContouringStlExporter_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/DualContouringStlExporter_tests.cpp
rename to gladius/tests/integrationtests/DualContouringStlExporter_tests.cpp
diff --git a/gladius/tests/unittests/GlobalMortonOctree_tests.cpp b/gladius/tests/integrationtests/GlobalMortonOctree_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/GlobalMortonOctree_tests.cpp
rename to gladius/tests/integrationtests/GlobalMortonOctree_tests.cpp
diff --git a/gladius/tests/unittests/HierarchicalDC_CompilationDebug_tests.cpp b/gladius/tests/integrationtests/HierarchicalDC_CompilationDebug_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/HierarchicalDC_CompilationDebug_tests.cpp
rename to gladius/tests/integrationtests/HierarchicalDC_CompilationDebug_tests.cpp
diff --git a/gladius/tests/unittests/HierarchicalDC_ExtractionStep_tests.cpp b/gladius/tests/integrationtests/HierarchicalDC_ExtractionStep_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/HierarchicalDC_ExtractionStep_tests.cpp
rename to gladius/tests/integrationtests/HierarchicalDC_ExtractionStep_tests.cpp
diff --git a/gladius/tests/unittests/HierarchicalDC_STLExport_tests.cpp b/gladius/tests/integrationtests/HierarchicalDC_STLExport_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/HierarchicalDC_STLExport_tests.cpp
rename to gladius/tests/integrationtests/HierarchicalDC_STLExport_tests.cpp
diff --git a/gladius/tests/unittests/HierarchicalDualContouring_tests.cpp b/gladius/tests/integrationtests/HierarchicalDualContouring_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/HierarchicalDualContouring_tests.cpp
rename to gladius/tests/integrationtests/HierarchicalDualContouring_tests.cpp
diff --git a/gladius/tests/unittests/ManifoldDualContouring_tests.cpp b/gladius/tests/integrationtests/ManifoldDualContouring_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/ManifoldDualContouring_tests.cpp
rename to gladius/tests/integrationtests/ManifoldDualContouring_tests.cpp
diff --git a/gladius/tests/unittests/MeshBaseline_tests.cpp b/gladius/tests/integrationtests/MeshBaseline_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/MeshBaseline_tests.cpp
rename to gladius/tests/integrationtests/MeshBaseline_tests.cpp
diff --git a/gladius/tests/unittests/MeshSdfPerformance_tests.cpp b/gladius/tests/integrationtests/MeshSdfPerformance_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/MeshSdfPerformance_tests.cpp
rename to gladius/tests/integrationtests/MeshSdfPerformance_tests.cpp
diff --git a/gladius/tests/unittests/NodeLayoutEngine_tests.cpp b/gladius/tests/integrationtests/NodeLayoutEngine_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/NodeLayoutEngine_tests.cpp
rename to gladius/tests/integrationtests/NodeLayoutEngine_tests.cpp
diff --git a/gladius/tests/unittests/PaletteExtractor_tests.cpp b/gladius/tests/integrationtests/PaletteExtractor_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/PaletteExtractor_tests.cpp
rename to gladius/tests/integrationtests/PaletteExtractor_tests.cpp
diff --git a/gladius/tests/unittests/ShellGenerator_tests.cpp b/gladius/tests/integrationtests/ShellGenerator_tests.cpp
similarity index 100%
rename from gladius/tests/unittests/ShellGenerator_tests.cpp
rename to gladius/tests/integrationtests/ShellGenerator_tests.cpp
